simulation/behavior_tree_plugin/src/vehicle/follow_trajectory_sequence/follow_polyline_trajectory_action.cpp

```diff
@@ -56,16 +56,37 @@ auto FollowPolylineTrajectoryAction::providedPorts() -> BT::PortsList
 
 auto FollowPolylineTrajectoryAction::tick() -> BT::NodeStatus
 {
+  auto getTargetSpeed = [&]() -> double {
+    if (target_speed.has_value()) {
+      return target_speed.value();
+    } else {
+      return entity_status->getTwist().linear.x;
+    }
+  };
+
   if (getBlackBoardValues();
       request != traffic_simulator::behavior::Request::FOLLOW_POLYLINE_TRAJECTORY or
       not getInput<decltype(polyline_trajectory)>("polyline_trajectory", polyline_trajectory) or
       not getInput<decltype(target_speed)>("target_speed", target_speed) or
       not polyline_trajectory) {
     return BT::NodeStatus::FAILURE;
+  } else if (std::isnan(entity_status->getTime())) {
+    THROW_SIMULATION_ERROR(
+      "Time in entity_status is NaN - FollowTrajectoryAction does not support such case.");
   } else if (
-    const auto updated_status = traffic_simulator::follow_trajectory::makeUpdatedStatus(
+    auto updated_status = traffic_simulator::follow_trajectory::makeUpdatedStatus(
       static_cast<traffic_simulator::EntityStatus>(*entity_status), *polyline_trajectory,
-      behavior_parameter, step_time, target_speed)) {
+      behavior_parameter, step_time, getTargetSpeed())) {
+    // matching distance has been set to 3.0 due to matching problems during lane changes
+    if (const auto lanelet_pose = hdmap_utils->toLaneletPose(
+          updated_status->pose, entity_status->getBoundingBox(), false, 3.0);
+        lanelet_pose) {
+      updated_status->lanelet_pose = lanelet_pose.value();
+      updated_status->lanelet_pose_valid = true;
+    } else {
+      updated_status->lanelet_pose_valid = false;
+    }
```

Comment:
It is not beautiful that the caller of makeUpdatedStatus has to set the lanelet_pose each time.
Please add HdMapUtils const& to the makeUpdatedStatus argument and change this process to the responsibility of makeUpdatedStatus.